networkx/algorithms/tree/mst.py

```diff
@@ -913,6 +915,20 @@ def spanning_tree_total_weight(G, weight):
                     )
                 return total
 
+    # Spanning tree for an empty graph is an empty graph
+    if G.number_of_nodes() == 0:
+        spanning_tree = nx.Graph()
+        return spanning_tree
+
+    # Spanning tree for a single node graph is that single node.
+    # A new graph is returned instead of returning G or its copy,
+    # in case G had a self loop
+    if G.number_of_nodes() == 1:
+        spanning_tree = nx.Graph()
+        spanning_tree.add_node(list(G.nodes)[0])
+        return spanning_tree
```

Comment:
An empty_graph has no edges -- but **can** have nodes. The **null graph** is the term you are looking for here: no nodes or edges.

Also, just to minimize the interruption of these corner cases on the code, let's make this more compact. Something like:
```suggestion
    if G.number_of_nodes() < 2:
        # no edges in the spanning tree
        return nx.empty_graph(G.nodes)
```

Reply:
Thanks, I'll make this change